### DISCUSSION TITLE
How `error_mapping` really works?

### DISCUSSION THREAD

**Discussion by GlucNAc** [symfony/symfony#50762] [Q&A]
Hello everyone,

I wanted to use the form option `error_mapping` as described in the doc: https://symfony.com/doc/current/reference/forms/types/form.html#error-mapping.

According to the doc, we can specify the unmapped property as the associative array key, and the target property to map the error to as the value of the associative array.

But when looking at the code: https://github.com/symfony/symfony/blob/27da66b55d49b58e16bc50c07b67c966d2d8ea2a/src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php#L128

it seems that it processes only the '.' keys of the array/nested arrays.

Thus I'm not sure about how to really use this functionality.

Could you help me with this please?

**Comment by ghost** [comment]
Docs: `You can construct nested property paths by concatenating them, separating properties by dots.`

The associative array key is for fields that are arrays.

Just try it empirically and see what (or if it) works. There is always a chance that documentation is wrong or just the wording is not perfect.
Also provide your use case if possible - it will be simpler for us to help.
